File: core/src/org/rexellentgames/dungeon/entity/item/accessory/equipable/DewVial.java
package org.rexellentgames.dungeon.entity.item.accessory.equipable;

import org.rexellentgames.dungeon.assets.Locale;
import org.rexellentgames.dungeon.entity.creature.player.Player;

public class DewVial extends Equipable {
	{
		name = Locale.get("dew_vial");
		description = Locale.get("dew_vial_desc");
		sprite = "item-dew_vial";
	}

	@Override
	public void onEquip() {
		super.onEquip();

		if (this.owner instanceof Player) {
			((Player) this.owner).healOnEnter = true;
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();

		if (this.owner instanceof Player) {
			((Player) this.owner).healOnEnter = false;
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/DewVial.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class DewVial : Equippable() {
	init {
		name = Locale.get("dew_vial")
		description = Locale.get("dew_vial_desc")
		sprite = "item-dew_vial"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.healOnEnter = true
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.healOnEnter = false
	}

	override fun getMaxLevel(): Int {
		return 6
	}
}